rotkehlchen/chain/ethereum/decoding/decoder.py

```diff
@@ -489,11 +491,11 @@ def _maybe_decode_erc20_721_transfer(
 
     def _maybe_enrich_transfers(  # pylint: disable=no-self-use
             self,
-            token: Optional[EthereumToken],  # pylint: disable=unused-argument
+            token: Optional[EthereumToken],
```

Comment:
No @yabirgb that's not what I meant.

The way you have implemented this now `_maybe_enrich_transfers()` gets called for every single event.

What would make more sense is to add this at the end of the `_maybe_decode_erc20_721_transfer()` and pass it the transfer itself itself of making it iterate the decoded events.

Does this make sense?

Reply:
Indeed passing the transfer itself makes everything beautiful